layout/_partials/post/post-footer.swig

```diff
@@ -22,8 +22,8 @@
     </div>
   {%- endif %}
 
-  {%- if theme.likely.enable or (theme.needmoreshare2.enable and theme.needmoreshare2.postbottom.enable) %}
-    <div class="social-share">
+  {%- if theme.likely.enable or (theme.needmoreshare2.enable and theme.needmoreshare2.postbottom.enable) or theme.sharejs.enable %}
+    <div class="post-social-share">
```

Comment:
No additional changes, add extra judgment for sharejs

Reply:
```
{%- if theme.sharejs.enable %}
  <div class="social-share" data-sites="{{ theme.sharejs.networks }}" data-wechat-qrcode-title="{{ __(theme.sharejs.wechat_qrcode.title) }}" data-wechat-qrcode-helper="{{ __(theme.sharejs.wechat_qrcode.prompt) }}"></div>
{%- endif %}
```

Or you can put options to front-end, we have global CONFIG, check head.swig.

Reply:
Previously I changed class `social-share` to `post-social-share` because share.js has hard-coded selector for `social-share`, which will result duplicated share component.

I just reviewed their code and found that `share-component` class is also supported, I'll change the classnames back.

Reply:
I changed class social-share to post-social-share because share.js has hard-coded selector for social-share, which is exactly the same and will definitely result in problems if multiple share components are enabled.

Reply:
I performed a global search to make sure that no css code selects for `social-share`, I think it's necessary to introduce this change.

Reply:
> ```
> {%- if theme.sharejs.enable %}
>   <div class="social-share" data-sites="{{ theme.sharejs.networks }}" data-wechat-qrcode-title="{{ __(theme.sharejs.wechat_qrcode.title) }}" data-wechat-qrcode-helper="{{ __(theme.sharejs.wechat_qrcode.prompt) }}"></div>
> {%- endif %}
> ```
> 
> Or you can put options to front-end, we have global CONFIG, check head.swig.

Is it suitable to put configuration scripts to `_partials/share/sharejs.swig`? Code may be injected to all pages even if no share components present, or should I put it under `_partials/post/post_footer.swig`?

Reply:
Forget this, you can check likely, css/html/js part is independent

Reply:
For now, keep these share services unify, If you have more ideas, open another PR when this merged.

Reply:
> I performed a global search to make sure that no css code selects for social-share, I think it's necessary to introduce this change.

`.social-share` will be blocked by adblock, but `.post-social-share` won't.